workflows/cloudify_system_workflows/deployment_environment.py

```diff
@@ -38,10 +38,17 @@ def generate_create_dep_tasks_graph(ctx,
             ctx.execute_task('cloudify_agent.operations.install_plugins',
                              kwargs={'plugins': plugins_to_install}))
 
-    sequence.add(
-        ctx.send_event('Starting deployment policy engine core'),
-        ctx.execute_task('riemann_controller.tasks.create',
-                         kwargs=policy_configuration or {}))
+    if (policy_configuration['groups'] and
```

Comment:
So here's the problem - what you wrote here works fine right now, however in the very near future we're going to introduce the notion of "scaling groups" in the DSL, and, well, long story short, there may be a blueprint which only has groups which use the specific policy type for scaling groups - in which case, we don't need a riemann core for the deployment after all.

rather than have the type hardcoded here, I think the condition here should be as follows: if there is *any* group which is mapped to a policy type *which is defined in policy_types*, we need to install riemann, otherwise don't.
i know this sounds a bit confusing right now, but basically this hardcoded type i'm referencing is a policy type which is actually not going to be defined in the "policy_types" dictionary - so that's why only if there's a group with an actual "custom" type, does riemann become relevant.